crates/bevy_sprite/src/texture_atlas.rs

```diff
@@ -1,94 +1,83 @@
-use crate::Anchor;
-use bevy_asset::Handle;
+use bevy_asset::{Assets, Handle};
 use bevy_ecs::component::Component;
 use bevy_math::{Rect, Vec2};
 use bevy_reflect::{FromReflect, Reflect, TypeUuid};
-use bevy_render::{color::Color, texture::Image};
+use bevy_render::texture::Image;
 use bevy_utils::HashMap;
 
-/// An atlas containing multiple textures (like a spritesheet or a tilemap).
+/// Maps a layout for a texture. Used with the [`TextureAtlas`] component it allows to
+/// either draw a specific area of the target texture, or to animate a sprite sheet.
+///
+/// Optionaly it can store a mapping from sub texture handles to the related area index (see
+/// [`TextureAtlasBuilder`]).
+///
 /// [Example usage animating sprite.](https://github.com/bevyengine/bevy/blob/latest/examples/2d/sprite_sheet.rs)
 /// [Example usage loading sprite sheet.](https://github.com/bevyengine/bevy/blob/latest/examples/2d/texture_atlas.rs)
+///
+/// [`TextureAtlasBuilder`]: crate::TextureAtlasBuilder
 #[derive(Reflect, FromReflect, Debug, Clone, TypeUuid)]
 #[uuid = "7233c597-ccfa-411f-bd59-9af349432ada"]
 #[reflect(Debug)]
-pub struct TextureAtlas {
-    /// The handle to the texture in which the sprites are stored
-    pub texture: Handle<Image>,
+pub struct TextureAtlasLayout {
     // TODO: add support to Uniforms derive to write dimensions and sprites to the same buffer
     pub size: Vec2,
     /// The specific areas of the atlas where each texture can be found
     pub textures: Vec<Rect>,
-    /// Mapping from texture handle to index
-    pub(crate) texture_handles: Option<HashMap<Handle<Image>, usize>>,
+    /// Texture handle to area index mapping. Set by [`TextureAtlasBuilder`].
+    ///
+    /// [`TextureAtlasBuilder`]: crate::TextureAtlasBuilder
+    pub texture_handles: Option<HashMap<Handle<Image>, usize>>,
```

Comment:
Usually I'm a bit iffy on Option<Collection>, but I think it makes sense here due to the need to handle a malformed and unset version here.

Reply:
Imo this component represents too many features, sprite sheet, atlas dynamic building, texture merging, etc. It should probably be reworked but this PR goal is specifically to unify Sprites and avoid custom handling of atlases